src/functions.php

```diff
@@ -377,13 +386,132 @@ function upload(string $xml, $config)
         'PhonebookImportFile' => [
             'type' => 'text/xml',
             'filename' => 'updatepb.xml',
-            'content' => $xml,
+            'content' => $xmlNewPhonebook->asXML(), // convert XML object to XML string
         ]
     ];
 
-    $result = $fritz->postFile($formfields, $filefields); // send the command
-
+    $result = $fritz->postFile($formfields, $filefields); // send the command to store new phonebook
     if (strpos($result, 'Das Telefonbuch der FRITZ!Box wurde wiederhergestellt') === false) {
         throw new \Exception('Upload failed');
     }
 }
+
+
+/**
+ * Downloads the old phone book from Fritzbox
+ *
+ * @param   Api   $fritz
+ * @param   array $config
+ * @return  SimpleXMLElement with the old existing phonebook
+ */
+function downloadPhonebook(Api $fritz, array $config)
+{
+    $formfields = [
+        'PhonebookId' => $config['phonebook']['id'],
+        'PhonebookExportName' => $config['phonebook']['name'],
+        'PhonebookExport' => "",
+    ];
+    $result = $fritz->postFile($formfields, []); // send the command to load existing phone book
+    if (substr($result, 0, 5) !== "<?xml") {
+        error_log("ERROR: Could not load old phonebook with ID=".$config['phonebook']['id']);
+        return new SimpleXMLElement("<xml><empty /></xml>");
```

Comment:
This feels odd. Better return "false"  instead of fake XML.

Reply:
We already log an error to console here. So, user is in control what happens.
The problem with "false" return is, that I promised this in the header: `@return  SimpleXMLElement`.
So returning false gives a phpstan error - so, I would keep it this way.

Reply:
Just return SimpleXMLElement|bool in the header